feature/subscriber-attributes/src/main/java/com/revenuecat/purchases/subscriberattributes/SubscriberAttributesManager.kt

```diff
@@ -106,4 +116,57 @@ class SubscriberAttributesManager(
 
         deviceCache.setAttributes(appUserID, attributesToBeSet)
     }
+
+    /**
+     * Collect gpsAdId and androidId automatically
+     *
+     * This method cannot be called in the main thread as it may block leading to ANRs.
+     * An IllegalStateException will be thrown if this is called on the main thread.
+     */
+    fun collectDeviceIdentifiers(
```

Comment:
should we just call the relevant logic on a background thread here then? that we don't have to think about it at all.

Reply:
also, kinda feels like this belongs in the config class? subscriberAttributesManager should only know what to do about subscriber attributes, but not how to get specific ones

Reply:
That makes sense. The Config class is actually a `Singleton`. The main purpose I thought about giving that class was to hold configuration parameters more than to fetch stuff from the system. In iOS, we have an AttributionFetcher. What do you think about creating that?

Reply:
makes sense to keep config for, well, config. 
the attributionFetcher sounds good 👍